Finished implementation of AliceODMContext:

caciobanu opened the pull request:
- added implementation of create, drop and empty schema
- default object manager service id is "doctrine_mongodb.odm.default_document_manager" not "doctrine.mongodb.entity_manager"

I mention that I was unable to add functional tests because of the current inheritance model; Behat doesn't allow registering twice the same step message.

theofidry commented:
> Behat doesn't allow registering twice the same step message.

Indeed that's a legitimate limitation of behat. That being said, we can still a function test by adding the `ODM` context to another test suite (and not to `default`). We could on the way name that test suite `omd_context` and renamed `default` to `orm_context`.

Then we can run Behat tests two times instead of once: one time for each test suite.

Aside from that, thanks very much for this work :)

caciobanu commented:
You are right, we could do a different test suite.
Could you provide me a configuration for Mongo from Travis-CI?

theofidry commented:
Sure:

``` yaml
services:
  - mongodb

before_install:
  - echo "extension = mongo.so" >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
```

If you still have some issues you can take a look at [HautelookAliceBundle Travis config](https://github.com/hautelook/AliceBundle/blob/1.x/.travis.yml)

caciobanu commented:
@theofidry I've implemented functional tests for ODM and added HHVM to the list of allowed failures because of the mongo requirement. Also PHP nightly fails because on 7.X the deprecated extension mongo.so it is not available.

theofidry commented:
Looks alright to me aside from a minor question.

> Also PHP nightly fails because on 7.X the deprecated extension mongo.so it is not available.

Indeed, Mongo is slow to catch up :(

caciobanu commented:
What's the question?

theofidry commented:
Oh my looks like I didn't leave the comment. I was asking if we could not use the same `.feature` file for both test suites to avoid to have multiple ones.

caciobanu commented:
I don't think that it will be possible as long as we have **@TestBundle/DataFixtures/_ODM_/dummy.yml** as paths to fixtures.

theofidry commented:
Ok that's fair, let's get this merged :)

theofidry commented:
Thanks @caciobanu

caciobanu commented:
Great!
